### PR TITLE
feat: Update authSig generation to use blockhash as its nonce

### PR DESCRIPTION
**Pull request by Adarsh-Kumar28**
# Description

Update `checkAndSignAuthMessage` & `signAndSaveAuthMessage` to use the blockhash from the client as its nonce.

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

General
- [X] I have performed a self-review of my code
- [X] I have fixed all grammar issues (can use an AI tool to check), and explanations are in active voice
- [X] I have checked the additions are concise
- [X] Language is consistent with existing documentation
- [X] My changes generate no new warnings

If I have added a new concept, I have
- [X] included a beginner friendly explanation
- [X] included a basic technical introduction and code sample

### CODE REVIEW

File: docs/sdk/authentication/auth-sig.md [hunk 1]
```diff
@@ -60,13 +65,18 @@ const authSig = await ethConnect.signAndSaveAuthMessage({
   account: walletAddress,
   chainId: 1,
   expiration: new Date(Date.now() + 1000 * 60 * 60 * 24).toISOString(),
```
Comment (joshLong145) [inline docs/sdk/authentication/auth-sig.md:67]: This exp is now too high for the network to accept. we should limit this to ~ `30 days`

File: docs/sdk/authentication/auth-sig.md [hunk 2]
```diff
@@ -39,9 +39,14 @@ import { checkAndSignAuthMessage } from '@lit-protocol/lit-node-client';
 
 const authSig = await checkAndSignAuthMessage({
   chain: "ethereum",
+  nonce,
 });
 ```
 
+:::note
+Be sure to use the latest blockhash from the `litNodeClient` as the nonce. You can get it from the `litNodeClient.getLatestBlockhash()`.
```
Comment (joshLong145) [inline docs/sdk/authentication/auth-sig.md:47]: can we also show how to do this if not using a web extension wallet?

**Review by joshLong145 — APPROVED** [review]

File: docs/sdk/authentication/auth-sig.md (hunk 1 above)
Comment (Adarsh-Kumar28) [reply docs/sdk/authentication/auth-sig.md:67]: It takes in milliseconds as its argument. So the exp is 24 hours from now

File: docs/sdk/authentication/auth-sig.md (hunk 2 above)
Comment (Adarsh-Kumar28) [reply docs/sdk/authentication/auth-sig.md:47]: I already have. Check the "Obtaining an AuthSig on the server-side" section below

**Review by joshLong145 — APPROVED** [review]

File: docs/sdk/authentication/auth-sig.md (hunk 2 above)
Comment (joshLong145) [reply docs/sdk/authentication/auth-sig.md:47]: I see that now, looks good, and approved.

**Pull request merged**